packages/launchpad/src/components/code/FileRow.cy.tsx

```diff
@@ -117,10 +117,10 @@ describe('FileRow', () => {
     cy.contains(changesRequiredDescription).should('be.visible')
     cy.get('pre').should('have.length', 2)
 
-    cy.get('.shiki').should('be.visible')
+    cy.get('div.rounded > div.text-left.cursor-text').should('be.visible')
```

Comment:
@senpl Could you explain the need for this change? If users are required to update their tests based on this change, this would be considered a breaking change.

Reply:
.shiki is not precise selector. In moment of search in this test there are two elements with this locator. One berly visible but still possible to locate.
This selector better explain what test intended to search.

Reply:
It might be breaking change for poorly created locators. Still it's closer to what user can actually see in browser. So it's probably good change. If I could locate element in browser then it's visible, but there should be some rules to determine if something is visible or not.

Reply:
Hi @senpl , 

There are two `.shiki` elements in this component. One of them is always hidden, and one of them that should show/hide on click. The hidden state for this element is caused by a parent element with maxHeight of 0, and overflow:hidden.

I do agree that the `.shiki` selector here should be more specific. I edited it to assert the expected visibility state of both elements:

```
    cy.get('[data-cy=valid] .shiki').should('not.be.visible')
    cy.get('[data-cy=changes] .shiki').should('be.visible')

    cy.contains('cypress/integration/command.js').click()

    cy.get('[data-cy=valid] .shiki').should('not.be.visible')
    cy.get('[data-cy=changes] .shiki').should('not.be.visible')
```

Unfortunately, while this test passes on the `develop`, it does not pass with the visibility changes in this branch. Can you look into why?

Thank you!

Reply:
I already did.
But this second locator is still visible. I could click it in browser and get it's locator. It's not big. It could to be very hard to click on it. Still it is visible. I do not see reason for Cypress to mark something as invisible when it is visible. It might produce warning that something is hardly visible, still if it's partly visible it should not expect that element to be not visible.

Reply:
As in pictures below .shrik width is bigger then element code below in moment of checking. This is why something that is  really invisible or not exist should be checked, not something that is visible in moment of click. Alternatively we could wait for this moment to become invisible, but this would make test slower and not as practical.
![chrome_rufv04rfEp](https://github.com/cypress-io/cypress/assets/5415941/76c456dc-be4a-4e3e-9819-e72b99b80bd5)
![chrome_tFds2UDHKr](https://github.com/cypress-io/cypress/assets/5415941/7a698838-5a00-4b37-9be5-0a4881200b51)